src/simulation.hpp

```diff
@@ -1084,6 +1084,15 @@ template <typename problem_t> void AMRSimulation<problem_t>::evolve()
 		computeBeforeTimestep();
 
 #if AMREX_SPACEDIM == 3
+		// TODO(cch): Need to take care of AMR subcycling
+		particleRegister_.destroyParticles(0, cur_time, dt_[0]);
+
+		// Stellar evolution and SN deposition; only apply to star particles
+		if (particleRegister_.HasStarParticles()) {
+			// TODO(cch): Need to take care of AMR subcycling
+			particleMeshInteraction(cur_time, dt_[0]);
```

Comment:
This will change the CFL-stable timestep. See my longer comment. I think this change needs very careful consideration.